src/controllers/auth.contoller.js

```diff
@@ -0,0 +1,129 @@
+const { userSchema } = require('../models/user');
+
+const { authServices } = require('../services/auth.services');
+const { jwtServices } = require('../services/jwt.services');
+const { tokenServices } = require('../services/token.services');
+const { usersServices } = require('../services/users.services');
+const {
+  sendActivationLink,
+  sendResetLink,
+} = require('../services/email.services');
+
+const { ApiError } = require('../exeptions/auth.error');
+
+const { checkPassword } = require('../utils/password/checkPassword');
+const { asyncHandler } = require('../utils/asyncHandler');
+
+const authValidation = async (req, _, next) => {
+  const { email, password } = req.body;
+
+  try {
+    await userSchema.validate({ email, password });
+  } catch (e) {
+    throw ApiError.badRequest(e.message);
+  }
+
+  req.userData = { email, password };
+  next();
+};
+
+const signUp = async (req, res) => {
+  const user = !!(await usersServices.getUserByEmail(req.userData.email));
+
+  if (user) {
+    throw ApiError.badRequest('User already exists');
+  }
+
+  const { email, activationToken } = await authServices.signUp(req.userData);
```

Comment:
The `signUp` function should validate the password against the rules specified in the task description before creating a user. Consider adding a password validation step here.